src/mmfatl.c

```diff
@@ -155,32 +113,64 @@ static void initBuffer(struct Buffer* buffer) {
 }
 
 /*!
+ * \brief check the message buffer for emptiness
+ *
+ * \param buffer [const, not null] the buffer to check for emptiness.
+ * \return true, iff the \ref buffer is in its initial state.
+ * \pre \ref initBuffer was called
+ */
+inline static bool isBufferEmpty(struct Buffer const* buffer) {
+  return buffer->begin == buffer->text;
+}
+
+/*!
+ * \brief last character in the message buffer
+ *
+ * Get the last stored character in the buffer.  Discarded characters due to
+ * overflow are ignored.  A returned NUL indicates the buffer is empty.
+ * \param buffer [const, not null] the buffer to investigate.
+ * \return the last character stored in buffer, or NUL iff empty.
+ * \pre \ref initBuffer was called
+ */
+static char getLastBufferedChar(struct Buffer const* buffer) {
+  return isBufferEmpty(buffer)? NUL : *(buffer->begin - 1);
+}
+
+/*!
+ * \brief check whether the buffer is overflown
+ *
  * \param buffer [const, not null] the buffer to check for overflow.
  * \return true, iff the current contents exceeds the capacity of the
  *   \ref buffer, so at least the terminating \ref NUL is cut off, maybe more.
+ * \pre \ref initBuffer was called
```

Comment:
```suggestion
 * \pre the buffer is initialized, by calling \ref initBuffer prior
```
Preconditions should assert facts about the current state of the program, not about things that happened in the past. (Copy paste to the other uses of this line)

Reply:
I know my English and the descriptions are imperfect.  This needs some iterations to get right.